### ISSUE TITLE
Additional CSS class(es) not working for Widget Group

### ISSUE THREAD

**Issue by glagonikas** [WordPress/gutenberg#38173]
### Description

As the title is saying, adding Additional CSS class(es) to a Widget Group doesn't appear to work.

### Step-by-step reproduction instructions

1. Go to Widgets
2. Add a Widget Group and insert a widget inside it
3. Go to "Additional CSS class(es)" of the Widget Group and enter a class name
4. Save and go to the front-end

**Expected Results**
The CSS class is added to the Widget Group

**Actual Results**
The CSS class is not added

### Environment info

- Wordpress Version 5.8.3
- Gutenberg Version 12.4.1
- TwentyTwentyOne or Storefront

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Confirmed, WordPress version 5.9-RC3-52619, with and without Gutenberg installed.

**Comment by noisysocks** [comment]
I don't think we can support this because the Widget Group block outputs something like this:

```html
<h2 class="widget-title">Title</h2>
<div class="wp-widget-group__inner-blocks"></div>
```

Note that there is no container markup for the classname to be added to.

The `<div class="widget">` container which appears in the frontend comes from the `WP_Widget_Block` widget, not the Widget Group block.

I am thinking we should set `supports.className` to `false` for the Widget Group block.

**Comment by glagonikas** [comment]
@noisysocks wouldn't it be better to wrap the Widget Group into a container?

I might be wrong, but the majority of blocks have a container, and if so, it's probably worth keeping them consistent?

Also, custom CSS on every block is quite important and having the ability to do it on a parent level, makes things easier.

**Comment by noisysocks** [comment]
If we wrap the `h2` and `div` in a container then there'll be two container divs in the resultant widget. This goes against the motivation of Widget Group which is to exactly match the markup of a traditional non-block widget.

What's your use case for adding a classname? Maybe there's another way.

**Comment by glagonikas** [comment]
one of our use cases is to style a group of widgets differently, for example having two widgets inside a grey box and one outside. Adding a class to each of the two widgets would result in grey bg behind them, but no bg on the margin between them.

Appreciate this is an edge case scenario, but in my mind (might be wrong or against Widget coding standards) every widget should be wrapped inside a container div so i can easily be targed via CSS? This way, you could put custom CSS to modify both the specific Widget Group's H2 as well as individual widgets inside?

So you could have
```
Widget Group <div>
- h2 (widget group title)
- Widget <div>
- - h2 (widget title)
- Widget <div>
- - h2 (widget title)
- ...
```

**Comment by SilverMira** [comment]
Would it not be possible for Widget Group's additional classes to target %2$s in `register_sidebar()` `before_widget`?, correct me if I'm wrong, is there a way currently for block widgets to accomplish this using the block editor? As I can see, the %2$s is being populated by `widget_block` from the internal `WP_Widget_Block` only, there's not an easy way to add values to this part to a specific widget from the editor.

```php
register_sidebar([
  //...
  'before_widget'  => '<section id="%1$s" class="widget %2$s">',
  'after_widget'   => '</section>',
  //...
]);
```

**Comment by eiskalteschatten** [comment]
I'm still having this issue over a year later with WordPress 6.2.2. Has anyone taken a look at it?

**Comment by tcgunel** [comment]
Wordpress 6.4.1 still same problem. Had to group the block and give additional css class to that group to make it work.

**Comment by jeremylind** [comment]
Still an issue in 6.5.3.

**Comment by dbarber-pace** [comment]
Still an issue in 6.7.1.

**Comment by vespino** [comment]
Still an issue in 6.9.
